src/compiler/program.ts

```diff
@@ -630,10 +630,18 @@ namespace ts {
             }
         }
 
+        function getUserDefinedLibFileName(options: CompilerOptions): string[] {

```

Comment:
we could change this to `getDefaultLibLocation() : string`, and make it optional on the interface. in case it is not defined, we call `getDefaultLibFileName` and use the folder name instead.